sdk/storage/storage-blob/src/BlobClient.ts

```diff
@@ -795,7 +808,11 @@ export class BlobClient extends StorageClient {
     } else {
       throw new Error("Expecting non-empty strings for containerName and blobName parameters");
     }
-    super(urlOrConnectionString, pipeline);
+
+    super(url, pipeline);
+    const names = this.getBlobAndContainerNamesFromUrl(url);
```

Comment:
Might be able to do destruction:
```js
{ this.blobName, this.containerName } = this.getBlobAndContainerNamesFromUrl(url);
```

Reply:
Tried. Couldn't do this. Had to rely on a new variable.

Reply:
maybe I didn't get the syntax right.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Assigning_to_new_variable_names

Reply:
Done.